tests/decay/test_viewer.py

```diff
@@ -49,26 +49,30 @@ def test_simple_decay_chain():
 
 
 checklist_decfiles = (
-    (DIR / "../data/test_Bc2BsPi_Bs2KK.dec", "B_c+sig"),
-    (DIR / "../data/test_Bd2DDst_Ds2DmPi0.dec", "B0sig"),
-    (DIR / "../data/test_Bd2DmTauNu_Dm23PiPi0_Tau2MuNu.dec", "B0sig"),
-    (DIR / "../data/test_Bd2DMuNu.dec", "anti-B0sig"),
-    (DIR / "../data/test_Bd2DstDst.dec", "anti-B0sig"),
-    (DIR / "../data/test_example_Dst.dec", "D*+"),
-    (DIR / "../data/test_Xicc2XicPiPi.dec", "Xi_cc+sig"),
+    (DIR / "../data/test_Bc2BsPi_Bs2KK.dec", "B_c+sig", False),
+    (DIR / "../data/test_Bd2DDst_Ds2DmPi0.dec", "B0sig", True),
+    (DIR / "../data/test_Bd2DmTauNu_Dm23PiPi0_Tau2MuNu.dec", "B0sig", False),
+    (DIR / "../data/test_Bd2DMuNu.dec", "anti-B0sig", False),
+    (DIR / "../data/test_Bd2DstDst.dec", "anti-B0sig", False),
+    (DIR / "../data/test_example_Dst.dec", "D*+", True),
+    (DIR / "../data/test_Xicc2XicPiPi.dec", "Xi_cc+sig", False),
 )
 
 
-@pytest.mark.parametrize("decfilepath,signal_mother", checklist_decfiles)
-def test_duplicate_arrows(decfilepath, signal_mother):
+@pytest.mark.parametrize("decfilepath,signal_mother,dup", checklist_decfiles)
+def test_duplicate_arrows(decfilepath, signal_mother, dup):
     """
     This test effectively checks whether any box node (node with subdecays)
     gets more than one arrow to it, which would show a bug
     in the creation of the DOT file recursively parsing the built decay chain.
     """
     p = DecFileParser(decfilepath, DIR / "../../src/decaylanguage/data/DECAY_LHCB.DEC")
 
-    p.parse()
+    if dup:
+        with pytest.warns(UserWarning, match="pi0"):
```

Comment:
What's the reason for this new warning showing up? Something to fix upstream?

Reply:
It's an old warning, we just weren't doing anything with it before. It's a duplicate particle warning, and I think it was intentional in these examples, they were supposed to have a duplicate particle definition warning?

Reply:
(I've been meaning to handle it for a while)

Reply:
Oh, now I remember - yes, I did add some checks of that kind because the parser is supposed to run for checks for silly user mistakes :D. Thanks as ever.